src/main/scala/dicom/DicomDeidentifyDictionary.scala

```diff
@@ -92,19 +97,16 @@ object DicomDeidentifyDictionary {
             tag = intTag,
             name = rowCellTexts(0),
             keyword = Keyword.valueOf(intTag),
-            action = rowCellTexts(4),
-            // todo: create Enum and use instead of Option[String]
-            retainUidsAction = Option(rowCellTexts(6)).filter(_.nonEmpty),
-            retainDevIdAction = Option(rowCellTexts(7)).filter(_.nonEmpty),
-            retainInstIdAction = Option(rowCellTexts(8)).filter(_.nonEmpty),
-            retainPatCharsAction = Option(rowCellTexts(9)).filter(_.nonEmpty),
-            retainLongFullDatesAction =
-              Option(rowCellTexts(10)).filter(_.nonEmpty),
-            retainLongModifDatesAction =
-              Option(rowCellTexts(11)).filter(_.nonEmpty),
-            cleanDescAction = Option(rowCellTexts(12)).filter(_.nonEmpty),
-            cleanStructContAction = Option(rowCellTexts(13)).filter(_.nonEmpty),
-            cleanGraphAction = Option(rowCellTexts(14)).filter(_.nonEmpty)
+            action = getActionCode(rowCellTexts(4)).getOrElse(ActionCode.X),
+            retainUidsAction = getActionCode(rowCellTexts(6)),
+            retainDevIdAction = getActionCode(rowCellTexts(7)),
+            retainInstIdAction = getActionCode(rowCellTexts(8)),
+            retainPatCharsAction = getActionCode(rowCellTexts(9)),
+            retainLongFullDatesAction = getActionCode(rowCellTexts(10)),
+            retainLongModifDatesAction = getActionCode(rowCellTexts(11)),
+            cleanDescAction = getActionCode(rowCellTexts(12)),
+            cleanStructContAction = getActionCode(rowCellTexts(13)),
+            cleanGraphAction = getActionCode(rowCellTexts(14))
```

Comment:
Changing those fields to a `deidOptionToAction: Map[DeidOption, ActionCode]` could prevent the boilerplate code?